File: 0121_Best Time to Buy and Sell Stock/BestTimetoBuyAndSellStock.cs
public class Solution {
    public int MaxProfit (int[] prices) {
        if (prices == null || prices.Length <= 1) return 0;
        int maxProfit = 0;
        int currentMin = prices[0];
        for (int i = 1; i < prices.Length; i++) {
            maxProfit = Math.Max (maxProfit, prices[i] - currentMin);
            currentMin = Math.Min (currentMin, prices[i]);
        }

        return maxProfit;
    }
}

File: 0121_Best Time to Buy and Sell Stock/BestTimetoBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices.length <= 1) return 0;
        int currMin = prices[0];
        int ans = 0;
        for(int i=1;i<prices.length;i++) {
            ans = Math.max(ans, prices[i] - currMin);
            currMin = Math.min(currMin, prices[i]);
        }
        
        return ans;
    }
}